File: problems/044-recent-submission-reuse/solutions/python/main.py
import sys


def input_tokens():
    for line in sys.stdin.buffer:
        yield from line.split()


tokens = input_tokens()
n, k = int(next(tokens)), int(next(tokens))
last_index = {}
hits = 0
for index in range(1, n + 1):
    fingerprint = next(tokens)
    previous = last_index.get(fingerprint)
    if previous is not None and index - previous <= k:
        hits += 1
    last_index[fingerprint] = index
print(hits)

File: problems/044-recent-submission-reuse/solutions/rust/main.rs
use std::collections::HashMap;
use std::io::{self, Read};

fn main() {
    let mut input = String::new();
    io::stdin().read_to_string(&mut input).unwrap();
    let mut tokens = input.split_whitespace();
    let n: usize = tokens.next().unwrap().parse().unwrap();
    let k: usize = tokens.next().unwrap().parse().unwrap();
    let mut last_index: HashMap<&str, usize> = HashMap::with_capacity(n * 2);
    let mut hits = 0usize;

    for index in 1..=n {
        let fingerprint = tokens.next().unwrap();
        if let Some(&previous) = last_index.get(fingerprint) {
            if index - previous <= k {
                hits += 1;
            }
        }
        last_index.insert(fingerprint, index);
    }
    println!("{}", hits);
}
